web/src/compose_validate.js

```diff
@@ -251,18 +251,25 @@ function show_wildcard_warnings(stream_id, $banner_container, wildcard_mention)
     // only show one error for any number of @all or @everyone mentions
     if ($banner_container.find(`.${CSS.escape(classname)}`).length === 0) {
         compose_banner.append_compose_banner_to_banner_list(wildcard_template, $banner_container);
+        add_wildcard_banner_shown_for_container($banner_container);
     }
-
-    user_acknowledged_wildcard = false;
 }
 
 export function clear_wildcard_warnings($banner_container) {
     const classname = compose_banner.CLASSNAMES.wildcard_warning;
     $banner_container.find(`.${CSS.escape(classname)}`).remove();
 }
 
-export function set_user_acknowledged_wildcard_flag(value) {
-    user_acknowledged_wildcard = value;
+export function add_wildcard_banner_shown_for_container($banner_container) {
+    shown_wildcard_banners_for_container.add(
+        $banner_container.closest("form").attr("id") ?? "send_message_form",
+    );
+}
+
+export function delete_wildcard_banner_shown_for_container($banner_container) {
+    shown_wildcard_banners_for_container.delete(
+        $banner_container.closest("form").attr("id") ?? "send_message_form",
+    );
```

Comment:
Maybe this ends up being fine because the forms will have similar roles, what happens if you start editing a message in the "all messages" view and then navigate to another view? I guess since the form ID essentially encodes the message ID, it may reuse the container ID as well ... but that might not be wrong since it'll always be a different message edit form for the same message?

Reply:
> it'll always be a different message edit form for the same message?

Yeah, that is right!